### PR TITLE
Update syn related crates to 1.0

### PR DESCRIPTION
**Pull request by Eijebong**

### CODE REVIEW

File: src/display.rs [hunk 1]
```diff
@@ -141,10 +146,10 @@ impl<'a, 'b> State<'a, 'b> {
 
         let fmt = match &list.nested[0] {
             NestedMeta::Meta(Meta::NameValue(MetaNameValue {
-                ident,
+                path,
                 lit: Lit::Str(s),
                 ..
-            })) if ident == "fmt" => s,
+            })) if path.segments.first().unwrap().ident == "fmt" => s,
```
Comment (JelteF) [inline src/display.rs]: Let's make that an expect

File: src/display.rs [hunk 2]
```diff
@@ -336,10 +342,10 @@ impl<'a, 'b> State<'a, 'b> {
         }
         let fmt_string = match &list.nested[0] {
             NestedMeta::Meta(Meta::NameValue(MetaNameValue {
-                ident,
+                path,
                 lit: Lit::Str(s),
                 ..
-            })) if ident == "fmt" => s.value(),
+            })) if path.segments.first().unwrap().ident == "fmt" => s.value(),
```
Comment (JelteF) [inline src/display.rs]: Again expect

**Comment by Eijebong** [comment]
There, replaced those unwraps by expects

**Comment by JelteF** [comment]
Thank you

**Comment by Eijebong** [comment]
When can we expect this to be released ? (I didn't bump the version here because you seem to have a procedure for releases)

**Comment by JelteF** [comment]
There's some more PRs I wanted to include in this release, given that this is a breaking change. I expect somewhere this month.

Do you need an official release sooner for some reason? If so I can easily create one.

**Comment by Eijebong** [comment]
Nah I guess somewhere this month is good enough. I still have a lot of work before this could be blocking a deduplication of syn in servo.

**Comment by nox** [comment]
> Do you need an official release sooner for some reason? If so I can easily create one.

Any ETA for the release? Just wondering, would be nice to get rid of syn 0.15 in the ecosystem.

**Comment by JelteF** [comment]
I've decided that I wanted the next release to be the last one before 1.0.0 of this library. So I wanted to clean up the the code and add some more useful features, so that people can try them out before making them stable for a long time. I expect this stuff to be finished around the end of October. If you need a release with syn 1.0.0 before that I can pretty easily create one with only that change though. If that's the case please let me know.

**Comment by nox** [comment]
Removing syn 0.15 entirely from the Servo tree is currently blocked on a complicated glutin bump so end of October should be ok. I'll ping again if we manage to take care of glutin faster.

**Pull request merged**